modulemd/v2/modulemd-util.c

```diff
@@ -172,6 +172,44 @@ modulemd_hash_table_sets_are_equal (GHashTable *a, GHashTable *b)
 }
 
 
+gboolean
+modulemd_hash_table_equals (GHashTable *a,
+                            GHashTable *b,
+                            GEqualFunc compare_func)
+{
+  g_autoptr (GPtrArray) a_1 = NULL;
+  g_autoptr (GPtrArray) b_1 = NULL;
+  gchar *key_1 = NULL;
+  gchar *key_2 = NULL;
+  int i = 0;
+
+  if (g_hash_table_size (a) != g_hash_table_size (b))
+    {
+      return FALSE;
+    }
+  a_1 = modulemd_ordered_str_keys (a, modulemd_strcmp_sort);
+  b_1 = modulemd_ordered_str_keys (b, modulemd_strcmp_sort);
+
+  for (i = 0; i < a_1->len; i++)
+    {
+      key_1 = g_ptr_array_index (a_1, i);
+      key_2 = g_ptr_array_index (b_1, i);
+
+      if (g_strcmp0 (key_1, key_2) != 0)
+        {
+          return FALSE;
+        }
+    }
+
+  if (!compare_func (a_1, b_1))
```

Comment:
You can shorten the rest of this function to:
```suggestion
  return compare_func (a_1, b_1);
```